src/NzbDrone.Core/Download/Clients/Porla/Porla.cs

```diff
@@ -0,0 +1,239 @@
+
+                var item = new DownloadClientItem
+                {
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this, false),
+                    DownloadId = torrent.InfoHash.Hash,
+                    OutputPath = outputPath + torrent.Name,
+                    RemainingSize = torrent.Total,
+                    RemainingTime = eta,
+                    Title = torrent.Name,
+                    TotalSize = torrent.Size,
+                    SeedRatio = torrent.Ratio
+                };
+
+                if (!string.IsNullOrEmpty(torrent.Error))
+                {
+                    item.Status = DownloadItemStatus.Warning;
+                    item.Message = torrent.Error;
+                }
+
+                _logger.Debug($"Torrent {torrent.InfoHash.Hash} state is {torrent.State}");
+
+                // deal with moving_storage???
+                if (torrent.FinishedDuration > 0)
+                {
+                    item.Status = DownloadItemStatus.Completed;
+                }
+
+                // else if (torrent.state == HadoukenTorrentState.QueuedForChecking)
+                // {
+                //    item.Status = DownloadItemStatus.Queued;
+                // }
+                // else if (torrent.State == HadoukenTorrentState.Paused)
+                // {
+                //    item.Status = DownloadItemStatus.Paused;
+                // }
+                else
+                {
+                    item.Status = DownloadItemStatus.Downloading;
+                }
+
+                // torrent.state exists, can't quite tell if it's passthrough of https://libtorrent.org/reference-Torrent_Status.html#state_t
+
+                item.CanMoveFiles = item.CanBeRemoved = true; // usure of the restrictions here
+
+                items.Add(item);
+            }
+
+            return items;
+        }
+
+        public override void RemoveItem(DownloadClientItem item, bool deleteData)
+        {
+            // Kinda sucks we don't have a `RemoveItems`, porla has a batch interface for removals
+            PorlaTorrent[] singleItem = { new PorlaTorrent(item.DownloadId.ToLower(), "") };
+            _proxy.RemoveTorrent(Settings, deleteData, singleItem);
+
+            // when do we set item.Removed ?
+        }
+
+        public override DownloadClientInfo GetStatus()
+        {
+            // var preset = _proxy.ListPresets(Settings);
+            // var sessettings = _proxy.GetSessionSettings(Settings);
+
+            var status = new DownloadClientInfo
+            {
+                IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                RemovesCompletedDownloads = false, // should be determined through Flags From  somewhere? I think session?
+            };
+
+            // the default directory value is stored in the presets values... i think?
+            // but will not always be active if not set
```

Comment:
If not set what happens is Sonarr doesn't send a path to download to?

Reply:
If we don't set a path, it doesn't want to accept it.
If the preset is setup on porla side, it's cool with it.

Reply:
I'll update this block when I get to it.

But yeah, if you have it set in the presets, it'll use that, else it expects you to set it in the request for the torrent.
If it is set in the presets, AND you set it in the torrent, it uses the request's value.